library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1488 @@
+                                                                      0,
+                                                                      rb,
+                                                                      0,
+                                                                      1)));
+                                }
+                            }
+                            else
+                            {
+                                for(int i = ii; i <= ii + isec - 1; i++)
+                                {
+                                    CHECK_INTERNAL_ROCBLAS_ERROR(
+                                        (rocblas_copy_template<NB, T>(handle,
+                                                                      jsec,
+                                                                      &c[i - 1 + (jj - 1) * ldc],
+                                                                      0,
+                                                                      ldc,
+                                                                      0,
+                                                                      &dt1[(i - ii) * ldt1],
+                                                                      0,
+                                                                      1,
+                                                                      0,
+                                                                      1)));
+                                }
+                            }
+                            //
+                            //                      T1 := alpha*T1*T2 + delta*T1, triangular matrix
+                            //                      multiply where the value of delta depends on
+                            //                      whether T2 stores a unit or non-unit triangular
+                            //                      block. Gamma and tsec are used to compensate for
+                            //                      a deficiency in DGEMV that appears if the second
+                            //                      dimension (tsec) is zero.
+                            //
+                            for(int i = ii; i <= ii + isec - 1; i++)
+                            {
+                                if(diag == rocblas_diagonal_non_unit)
+                                {
+                                    delta                  = *alpha;
+                                    bool pointer_mode_host = false;
+                                    if(handle->pointer_mode == rocblas_pointer_mode_host)
+                                    {
+                                        pointer_mode_host = true;
+                                        CHECK_INTERNAL_ROCBLAS_ERROR((rocblas_set_pointer_mode(
+                                            handle, rocblas_pointer_mode_device)));
```

Comment:
We should call `handle->push_pointer_mode()` instead of `rocblas_set_pointer_mode` if possible. It changes the current mode and restores the object returned by `push_pointer_mode()` is destroyed. It is automatically exception-safe.

Reply:
Changed to use handle->push_pointer_mode()